projects/dex-ui/src/components/ConnectWalletButton.tsx

```diff
@@ -0,0 +1,20 @@
+import { ConnectKitButton } from "connectkit";
+import React, { ComponentProps } from "react";
+import { Button } from "src/components/Swap/Button";
+import { useAccount } from "wagmi";
+
+type ActionWalletButtonProps = ComponentProps<typeof Button>;
+
+export const ActionWalletButton = (props: ActionWalletButtonProps) => {
+  const { address } = useAccount();
+
+  return !address ? (
+    <ConnectKitButton.Custom>
+      {({ show }) => {
+        return <Button onClick={show} label="Connect Wallet" />;
+      }}
+    </ConnectKitButton.Custom>
+  ) : (
+    <Button {...props} />
+  );
```

Comment:
Not a _required_ change, up to you if you want to implement/have time, but it might have been cleaner to pass the actual button (to be displayed if the wallet is connected) as a  prop, instead of hard coding it in this component... that way the consumer would have flexibility to change it however they want:

```
 <ActionWalletButton>
    <Button myprops='here' />
 </ActionWalletButton>
...
 <ActionWalletButton>
    <OtherButton myprops='there' />
 </ActionWalletButton>

```